test/extended/build_context_dir_test.go

```diff
@@ -0,0 +1,68 @@
+package extended
+
+import (
+	"testing"
+	"time"
+
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	buildapi "github.com/openshift/origin/pkg/build/api"
+	"github.com/openshift/origin/test/util"
+)
+
+func init() {
+	util.RequireDocker()
+	util.RequireServer()
+}
+
+func TestBuildContextDir(t *testing.T) {
+	build := mockBuildWithContextDir()
+	client := util.GetClient()
+	if _, err := client.Builds("contextdir").Create(build); err != nil {
```

Comment:
maybe a random namespace would be better than fixed "contextdir". That way in theory you could run this test multiple times in parallel if that's what you want to do.